File: yunva17/프로그래머스/Lv2/올바른_괄호.js
function solution(s) {
  var answer = true;
  var count = 0;
  var stack = [];

  for (var i = 0; i < s.length; i++) {
    if (s[i] === '(') {
      stack.push('(');
      count += 1;
    } else {
      stack.pop();
      count -= 1;
    }
  }

  if (stack.length > 0 || count !== 0) {
    answer = false;
  }

  return answer;
}

File: overthename/Programmers/스택&큐/올바른괄호.java
import java.util.*;
class Solution {
	boolean solution(String s) {
		boolean answer = true;

		Stack<Character> st = new Stack<>();
		for(int i=0; i<s.length();i++){
			if(s.charAt(i)=='('){
				st.push('(');
			}else{

				if(st.isEmpty()){
					return false;
				}else{
					st.pop();
				}
			}
		}
		if(st.size()!=0){
			return false;
		}

		return true;
	}
}